src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1897,7 +1919,7 @@ public override BoundNode VisitConversion(BoundConversion node)
 
                 if (node.ExplicitCastInCode &&
                     !node.IsExplicitlyNullable &&
-                    targetType.IsReferenceType &&
+                    (targetType.IsReferenceType || IsUnconstrainedTypeParameter(targetType)) &&
                     (operandType?.IsNullable == true || (operandType is null && operand.IsLiteralNullOrDefault())))
```

Comment:
>(operandType?.IsNullable == true || (operandType is null && operand.IsLiteralNullOrDefault())) [](start = 20, length = 94)

Should we check for an unconstrained type parameter as an operand?

Reply:
Yes, good catch. Added.

---
In reply to: [177552500](https://github.com/dotnet/roslyn/pull/25733#discussion_r177552500) [](ancestors = 177552500)